cfme/middleware/jdr_report.py

```diff
@@ -0,0 +1,141 @@
+import re
+
+from datetime import datetime
+
+from cfme.utils import attributize_string
+from cfme.utils.appliance import current_appliance
+
+
+def _db_select_query(server):
+    """column order: `id`, `queued_at`, `requesting_user`, `status`"""
+    t_ms = current_appliance.db.client['middleware_servers']
+    t_mdr = current_appliance.db.client['middleware_diagnostic_reports']
+    query = current_appliance.db.client.session.query(
+        t_mdr.id, t_mdr.queued_at,
+        t_mdr.requesting_user.label('username'),
+        t_mdr.status)\
+        .join(t_mdr, t_mdr.middleware_server_id == t_ms.id)
+    if server:
+        query = query.filter(t_ms.name == server.name)
+    return query
+
+
+class JDRReport():
+    """
+    JDRReport class provides actions and details of JDR Reports List
+     on Middleware Server page.
+    Class method available to get existing JDR list for provided Server
+
+    Args:
+        username: user queued the JDR Report
+        status: status of the JDR Report
+        server: Server for which report is generated
+        filename: name of the JDR Report file
+        size: size of the JDR Report file
+        queued_at: queued time of the JDR Report
+        db_id: database row id of JDR Report
+
+    Usage:
+
+        myreport = JDRReport(username='admin', status='Ready', server=eap_server)
+        myreport.download()
+
+        myreports = JDRReport.reports()
```

Comment:
**REQUIRED**
```python
Usage:

.. code-block:: python

    myreport = JDRReport(username='admin', status='Ready', server=eap_server)
    myreport.download()

    myreports = JDRReport.reports()
```

Reply:
have changed it accordingly, not sure why it still there